keystore/src/connection/platform/generic/migrations/V12__unique_id_constraints.sql

```diff
@@ -10,4 +10,18 @@ SELECT lower(hex(pk)), pk, sk FROM mls_encryption_keypairs;
 
 DROP TABLE mls_encryption_keypairs;
 
-ALTER TABLE mls_encryption_keypairs_new RENAME TO mls_encryption_keypairs;
\ No newline at end of file
+ALTER TABLE mls_encryption_keypairs_new RENAME TO mls_encryption_keypairs;
+
+---- mls_epoch_encryption_keypairs ----
+CREATE TABLE mls_epoch_encryption_keypairs_new (
+    id_hex TEXT UNIQUE,
+    id BLOB,
+    keypairs BLOB
+);
+
+INSERT INTO mls_epoch_encryption_keypairs_new (id_hex, id, keypairs)
+SELECT lower(hex(id)), id, keypairs FROM mls_epoch_encryption_keypairs;
```

Comment:
This seems sort-of wasteful. We have a blob `id` and we're now inserting its text represented, as a sequence of hex characters. For each original byte, we will have a pair of hex characters, each needing a byte (assuming UTF-8 encoding).

So we've gone from N to N + 2N == 3N bytes.